### PR TITLE
InstallAppRequestValidator update v1.0

### PR DESCRIPTION
**Pull request by a5x55555**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1027

### CODE REVIEW

File: src/AdminService.Validation/InstallAppRequestValidator.cs [hunk 1]
```diff
@@ -57,10 +58,10 @@ public class InstallAppRequestValidator : AbstractValidator<InstallAppRequest>,
             .Cascade(CascadeMode.Stop)
             .NotEmpty().WithMessage("Login can't be empty.")
             .Must(x => char.IsLetter(x[0])).WithMessage("Login must start with a letter.")
-            .MinimumLength(3).WithMessage("Login is too short.")
+            .MinimumLength(6).WithMessage("Login is too short.")
```
Comment (MariyaSin) [inline src/AdminService.Validation/InstallAppRequestValidator.cs]: min 5

File: src/AdminService.Validation/InstallAppRequestValidator.cs [hunk 2]
```diff
@@ -57,10 +58,10 @@ public class InstallAppRequestValidator : AbstractValidator<InstallAppRequest>,
             .Cascade(CascadeMode.Stop)
             .NotEmpty().WithMessage("Login can't be empty.")
             .Must(x => char.IsLetter(x[0])).WithMessage("Login must start with a letter.")
```
Comment (MariyaSin) [inline src/AdminService.Validation/InstallAppRequestValidator.cs]: remove

File: src/AdminService.Validation/InstallAppRequestValidator.cs [hunk 3]
```diff
@@ -57,10 +58,10 @@ public class InstallAppRequestValidator : AbstractValidator<InstallAppRequest>,
             .Cascade(CascadeMode.Stop)
             .NotEmpty().WithMessage("Login can't be empty.")
             .Must(x => char.IsLetter(x[0])).WithMessage("Login must start with a letter.")
-            .MinimumLength(3).WithMessage("Login is too short.")
+            .MinimumLength(6).WithMessage("Login is too short.")
             .MaximumLength(15).WithMessage("Login is too long.")
-            .Must(x => x.All(char.IsLetterOrDigit))
-            .WithMessage("Login must contain only letters or digits.");
+            .Must(x => LoginRegex.IsMatch(x))
+            .WithMessage("Login must contain only Latin letters or digits.");
```
Comment (MariyaSin) [inline src/AdminService.Validation/InstallAppRequestValidator.cs]: latin letters AND digits

Comment (ndoriomedov) [reply src/AdminService.Validation/InstallAppRequestValidator.cs]: updates - login must contains only letters or letters and digits

File: src/AdminService.Validation/InstallAppRequestValidator.cs [hunk 4]
```diff
@@ -12,6 +12,7 @@ public class InstallAppRequestValidator : AbstractValidator<InstallAppRequest>,
   {
     private static Regex NameRegex = new(@"^([a-zA-Zа-яА-ЯёЁ]+|[a-zA-Zа-яА-ЯёЁ]+[-|']?[a-zA-Zа-яА-ЯёЁ]+|[a-zA-Zа-яА-ЯёЁ]+[-|']?[a-zA-Zа-яА-ЯёЁ]+[-|']?[a-zA-Zа-яА-ЯёЁ]+)$");
     private static Regex PasswordRegex = new(@"(?=.*[.,:;?!*+%\-<>@[\]{}/\\_{}$#])");
+    private static Regex LoginRegex = new(@"^[a-zA-Z0-9]+$");
```
Comment (ndoriomedov) [inline src/AdminService.Validation/InstallAppRequestValidator.cs]: i think digits only login will be valid in this regex

File: src/AdminService.Validation/InstallAppRequestValidator.cs (hunk 3 above)
Comment (a5x55555) [reply src/AdminService.Validation/InstallAppRequestValidator.cs]: done

**Review by ndoriomedov — APPROVED** [review]

File: src/AdminService.Validation/InstallAppRequestValidator.cs [hunk 5]
```diff
@@ -56,11 +57,10 @@ public class InstallAppRequestValidator : AbstractValidator<InstallAppRequest>,
...
             .Cascade(CascadeMode.Stop)
             .NotEmpty().WithMessage("Login can't be empty.")
-            .Must(x => char.IsLetter(x[0])).WithMessage("Login must start with a letter.")
-            .MinimumLength(3).WithMessage("Login is too short.")
+            .MinimumLength(5).WithMessage("Login is too short.")
             .MaximumLength(15).WithMessage("Login is too long.")
-            .Must(x => x.All(char.IsLetterOrDigit))
-            .WithMessage("Login must contain only letters or digits.");
+            .Must(x => LoginRegex.IsMatch(x))
+            .WithMessage("Login must contain only Latin letters and digits.");
```
Comment (MariyaSin) [inline src/AdminService.Validation/InstallAppRequestValidator.cs]: "Login must contain only Latin letters and digits or only Latin letters."

**Review by MariyaSin — APPROVED** [review]

**Review by vorobevamv — APPROVED** [review]

**Review by EAndrey1091 — APPROVED** [review]

**Pull request merged**